### ISSUE TITLE
Read this before downloading from Mildom

### ISSUE THREAD

**Issue by nao20010128nao** [yt-dlp/yt-dlp#251]
# TL;DR
Notify @nao20010128nao before downloading VODs a lot! Bandwidth exceeded!

(copied from https://github.com/nao20010128nao/ytdl-patched/issues/22)

# Mildom extractor use proxy at Vercel
A game-oriented live website, [Mildom](https://mildom.com) serves the video over HTTP with HLS protocol.    
Since all HLS playlist/segments requires to append extra queries/headers before requesting to Mildom servers (usually, this is done in HLS playlist, but Mildom don't!), proxying required.    
The proxy is hosted at [Vercel](https://vercel.com/) as [bookish-octo-barnacle.vercel.app](https://bookish-octo-barnacle.vercel.app) under my account.

# Vercel has a limit
According to their [Fair Use Policy](https://vercel.com/docs/platform/fair-use-policy), Hobby plan has bandwidth limit of 100GB.    
The proxy, bookish-octo-barnacle, has served 101GB from April 1st, as of writing. Most of them caused by Mildom requests.

# Later
First of all, notify @nao20010128nao first, before you download very much. Can be Issue at [my project](https://github.com/nao20010128nao/ytdl-patched/issues/new/choose), or via Email to me. You can use your language if DeepL or Google Translate allows. I can host proxy at my server (or somewhere else) as an alternative for you.    
Endpoint will be temporarily disabled at 15:00 JST of April 19th (or 6:00 of same day in GMT). Download will fail until end of April. (schedule may change)

**Comment by nao20010128nao** [comment]
(日本語)
# 要約
VODをダウンロードする前に@nao20010128naoに予告してくれ！！！！帯域オーバーだよ！！！！！！

# Mildom extractorはVercelに置いてあるプロキシで動いている
[Mildom](https://mildom.com)はHLSで動画を配信しているのはご存知のことかと思います(ダウンロード開始時にfragmentsだったりffmpegが起動するのはそれが理由です)。    
これらのHLSプレイリスト/セグメントは、HTTPリクエストを投げる前に特定のクエリとヘッダを付加しなければならず、このためだけに@nao20010128naoのアカウントで[Vercelでプロキシを動かしています](https://bookish-octo-barnacle.vercel.app)。

# Vercelにも限りがある
Vercelの[Fair Use Policy](https://vercel.com/docs/platform/fair-use-policy)によると、Hobbyプランでは100GBの帯域制限があります。    
プロキシ(bookish-octo-barnacle)は4/1からこのIssueを立てた時点で累計101GBを配信しており、そのトラフィックのほとんどがMildomのリクエストです。

# お願いと今後
何よりも、大量ダウンロードするなら予告して下さい。[**@nao20010128naoのプロジェクト**](https://github.com/nao20010128nao/ytdl-patched/issues/new/choose)のIssueでも良いです。@nao20010128naoのメールでも良いです。ただしyt-dlpには送らないでください。DeepLかGoogle翻訳が使えるなら自国語で書いていいです。何か対応策を提示します。    
なお、プロキシは4/19 日本時間15:00を以って、月末まで一時的に封印します。(スケジュールは変更になる可能性があります。)

**Comment by nao20010128nao** [comment]
@pukkandan As I requested you in Discord, can you please pin this issue at least until end of April?

**Comment by fstirlitz** [comment]
Why even have a proxy? What does the proxy do? Surely you don’t need a proxy when opening the URL in the browser. Why can’t the fixups be done locally, in-process? Unless this is done for IP spoofing, this is a gratuitous dependency on a third-party service. And even if it’s spoofing, it’s still out of scope. This should not have been merged in the first place.

**Comment by nao20010128nao** [comment]
> Why even have a proxy? What does the proxy do? Surely you don’t need a proxy when opening the URL in the browser. Why can’t the fixups be done locally, in-process? Unless this is done for IP spoofing, this is a gratuitous dependency on a third-party service. And even if it’s spoofing, it’s still out of scope. This should not have been merged in the first place.

As I write in the Issue, HLS playlist for Mildom live does not contain required queries. Proxy is needed to fix it.
